### PR TITLE
Infiltrator Radio, adds new antag hud, fixes them spawning with white hair

### PR DESCRIPTION
**Pull request by SmoSmoSmoSmok**
## About The Pull Request

 adds the infiltrator radio to the game. an item for infiltrators allows them to  CLAIM a reward for every objective they complete. these rewards are mostly gadget uplink items. After every objective has been claimed, the infiltrator in question will gain a hijack objective. still a WORK IN PROGRESS so dont review this now

https://user-images.githubusercontent.com/95004236/177129788-a29cbadc-07bc-46b4-b673-9316a86cf39a.mp4

## Changelog

:cl:
/:cl:

### CODE REVIEW

File: fulp_modules/features/antagonists/infiltrators/infil_items.dm [hunk 1]
```diff
@@ -63,3 +63,83 @@
...
+	/obj/item/card/emag,
+	/obj/item/card/emag/doorjack,
+	/obj/item/grenade/syndieminibomb,
+	/obj/item/grenade/c4/x4,
+	/obj/item/pen/edagger,
+	/obj/item/guardiancreator/tech/choose/traitor,
+	/obj/item/jammer,
+	/obj/item/reagent_containers/hypospray/medipen/stimulants,
+	/obj/item/storage/box/syndie_kit/imp_stealth,
+	)
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/infiltrators/infil_items.dm:83]:
```suggestion
	var/reward_items = list(
		/obj/item/card/emag,
		/obj/item/card/emag/doorjack,
		/obj/item/grenade/syndieminibomb,
		/obj/item/grenade/c4/x4,
		/obj/item/pen/edagger,
		/obj/item/guardiancreator/tech/choose/traitor,
		/obj/item/jammer,
		/obj/item/reagent_containers/hypospray/medipen/stimulants,
		/obj/item/storage/box/syndie_kit/imp_stealth,
	)
```

Comment (JohnFulpWillard) [reply fulp_modules/features/antagonists/infiltrators/infil_items.dm:83]: and unique vars should always be under inherited ones

File: fulp_modules/features/antagonists/infiltrators/infil_items.dm [hunk 2]
```diff
@@ -63,3 +63,83 @@
...
+	desc = "How is this thing running without a battery?"
+	icon = 'fulp_modules/features/antagonists/infiltrators/icons/infils.dmi'
+	icon_state = "infiltrator_radio"
+	w_class = WEIGHT_CLASS_SMALL
+	///has the player claimed all his objectives' rewards?
+	var/objectives_complete = FALSE
+
+/obj/item/infiltrator_radio/proc/check_reward_status(mob/living/user)
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/infiltrators/infil_items.dm:91]: What if all infiltrator objectives send a signal to show it's completed an objective, that way we dont have to check every single ui_data for every objective in this guy's mind to see if any are done?

File: fulp_modules/features/antagonists/infiltrators/infil_items.dm [hunk 3]
```diff
@@ -63,3 +63,83 @@
...
+	if(.)
+		return
+	switch(action)
+		if("fuck")
+			if(check_reward_status(usr))
+				give_reward(usr)
+	return
+
+
+/obj/item/infiltrator_radio/Initialize(mapload)
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/infiltrators/infil_items.dm:143]: Initialize should be near the very start of the file

File: tgui/packages/fulpui-patches/interfaces/InfilRadio.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+              grand theft auto their emergency escape shuttle for spares.
+            </Box>
+          )|| (data.check ? 'Your work has satisfied our investors.'
+            : 'Prove your worth by completing an objective.') }
+        </Section>
+        {!completed &&(
+          <Section>
+            <Button
+              content={check ? "Claim Reward" : "Reward Unavailable"}
+              onClick={() => act('fuck')} />
```
Comment (JohnFulpWillard) [inline tgui/packages/fulpui-patches/interfaces/InfilRadio.js:31]:
```suggestion
              onClick={() => act('claim_reward')} />
```

File: fulp_modules/features/antagonists/infiltrators/infil_items.dm [hunk 5]
```diff
@@ -63,3 +63,83 @@
...
+	data["check"] = check_reward_status(user)
+	data["completed"] = objectives_complete
+	return data
+
+/obj/item/infiltrator_radio/ui_act(action, params)
+	. = ..()
+	if(.)
+		return
+	switch(action)
+		if("fuck")
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/infiltrators/infil_items.dm:137]:
```suggestion
		if("claim_reward")
```

File: fulp_modules/features/antagonists/infiltrators/infil_items.dm [hunk 6]
```diff
@@ -63,3 +63,83 @@
...
+	var/datum/objective/addobj = check_reward_status(user)
+	checked_objectives += addobj.explanation_text
+	var/reward = pick(reward_items)
+	podspawn(list(
+		"target" = get_turf(user),
+		"style" = STYLE_SYNDICATE,
+		"spawn" = reward,
+		))
+	reward_items -= reward //cant get duplicate rewards
+	var/list/bitch = user.mind.get_all_objectives()
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/infiltrators/infil_items.dm:107]: keep this please

File: fulp_modules/features/antagonists/infiltrators/infiltrator.dm [hunk 7]
```diff
@@ -53,6 +54,7 @@
 	backpack_contents = list(
 		/obj/item/storage/box/survival/syndie = 1,
 		/obj/item/knife/combat/survival = 1,
+		/obj/item/infiltrator_radio =1,
```
Comment (OneAsianTortoise) [inline fulp_modules/features/antagonists/infiltrators/infiltrator.dm:57]:
```suggestion
		/obj/item/infiltrator_radio = 1,
```

Comment (SmoSmoSmoSmok) [reply fulp_modules/features/antagonists/infiltrators/infiltrator.dm:57]: Shut the fuck up

Comment (SmoSmoSmoSmok) [reply fulp_modules/features/antagonists/infiltrators/infiltrator.dm:57]: Who you think you are? You should be kissing my ass right now

Comment (OneAsianTortoise) [reply fulp_modules/features/antagonists/infiltrators/infiltrator.dm]: U mad cuz of my awesome review

**Comment by Pepsilawn** [comment]
good 1st pr

**Pull request merged**